packages/utils/src/CopyClipboardUtils.ts

```diff
@@ -0,0 +1,43 @@
+class CopyClipboardUtils {
+  /**
+   * Copy the passed in text to the clipboard.
+   * @param text The text to copy
+   * @returns Promise Resolved on success, rejected on failure
+   */
+  static copyToClipboard(text: string): Promise<void> {
+    const { clipboard } = navigator;
+    if (clipboard === undefined) {
+      CopyClipboardUtils.copyToClipboardExecCommand(text);
+      return Promise.resolve();
+    }
+    return navigator.clipboard.writeText(text).catch(() => {
+      CopyClipboardUtils.copyToClipboardExecCommand(text);
+    });
+  }
```

Comment:
This could be written as an `async` function - right now `copyToClipbardExecCommand` could throw before a promise is returned and that would screw up a consumer. Something like 
```
  static async copyToClipboard(text: string): Promise<void> {
    const { clipboard } = navigator;
    if (clipboard !== undefined) {
      try {
        return navigator.clipboard.writeText(text);
      } catch (e) {
        // fallback to copyToClipboardExecCommand
      }
    }
    copyToClipboardExecCommand(text);
  }
```